File: app/src/main/java/com/example/administrator/character/StoresEmployee.java
package com.example.administrator.character;


import com.example.administrator.item.Item;
import com.example.administrator.item.SellItem;
import com.example.administrator.buildings.Building;
import com.example.administrator.utils.Tools;

public class StoresEmployee extends Employee {

    @Override
    public boolean work(Item item) {
        if (!(item instanceof SellItem)) return false;
        item.setTotal(item.getTotal()-1);
        Building workSpace = Building.findWorkSpace(getMaster());
        Character master = Tools.findMaster(workSpace.getMaster(),getCharacters());
        master.setMoney(master.getMoney() + ((SellItem)item).getSellPrice());
        return true;
    }

}

File: app/src/main/java/com/example/administrator/character/StoresEmployee.kt
package com.example.administrator.character


import com.example.administrator.item.Item
import com.example.administrator.item.SellItem
import com.example.administrator.buildings.Building
import com.example.administrator.utils.Tools

class StoresEmployee : Employee() {



    override fun work(item: Item,customer: Character){
        if (item !is SellItem) return
        item.total = item.total - 1
        val workSpace = Building.findWorkSpace(master)
        val master =  Character.characters[workSpace.master]
        master?.let { it.money += item.sellPrice }
        customer.money-=item.sellPrice
    }

    override fun receive(item: Item): Boolean = item is SellItem

}
